ur_robot_driver/src/hardware_interface.cpp

```diff
@@ -933,56 +997,96 @@ hardware_interface::return_type URPositionHardwareInterface::prepare_command_mod
-        if (control_modes[i] == hardware_interface::HW_IF_POSITION) {
-          control_modes[i] = "UNDEFINED";
-        }
+        stop_modes_[i].push_back(StoppingInterface::STOP_POSITION);
+        control_modes[i].erase(
+            std::remove_if(control_modes[i].begin(), control_modes[i].end(),
+                           [](const std::string& item) { return item == hardware_interface::HW_IF_POSITION; }),
+            control_modes[i].end());
       }
       if (key == info_.joints[i].name + "/" + hardware_interface::HW_IF_VELOCITY) {
-        stop_modes_.push_back(StoppingInterface::STOP_VELOCITY);
-        if (control_modes[i] == hardware_interface::HW_IF_VELOCITY) {
-          control_modes[i] = "UNDEFINED";
-        }
+        stop_modes_[i].push_back(StoppingInterface::STOP_VELOCITY);
+        control_modes[i].erase(
+            std::remove_if(control_modes[i].begin(), control_modes[i].end(),
+                           [](const std::string& item) { return item == hardware_interface::HW_IF_VELOCITY; }),
+            control_modes[i].end());
+      }
+      if (key == tf_prefix + FORCE_MODE_GPIO + "/disable_cmd") {
+        stop_modes_[i].push_back(StoppingInterface::STOP_FORCE_MODE);
+        control_modes[i].erase(std::remove_if(control_modes[i].begin(), control_modes[i].end(),
+                                              [&](const std::string& item) { return item == FORCE_MODE_GPIO; }),
+                               control_modes[i].end());
       }
       if (key == tf_prefix + PASSTHROUGH_GPIO + "/setpoint_positions_" + std::to_string(i)) {
-        stop_modes_.push_back(StoppingInterface::STOP_PASSTHROUGH);
-        if (control_modes[i] == PASSTHROUGH_GPIO) {
-          control_modes[i] = "UNDEFINED";
-        }
+        stop_modes_[i].push_back(StoppingInterface::STOP_PASSTHROUGH);
+        control_modes[i].erase(std::remove_if(control_modes[i].begin(), control_modes[i].end(),
+                                              [&](const std::string& item) { return item == PASSTHROUGH_GPIO; }),
+                               control_modes[i].end());
       }
     }
   }
 
   // Do not start conflicting controllers
-  if (std::any_of(start_modes_.begin(), start_modes_.end(),
+  if (std::any_of(start_modes_[0].begin(), start_modes_[0].end(),
                   [this](auto& item) { return (item == PASSTHROUGH_GPIO); }) &&
-      (std::any_of(start_modes_.begin(), start_modes_.end(),
+      (std::any_of(start_modes_[0].begin(), start_modes_[0].end(),
+                   [](auto& item) {
+                     return (item == hardware_interface::HW_IF_VELOCITY || item == hardware_interface::HW_IF_POSITION);
+                   }) ||
+       std::any_of(control_modes[0].begin(), control_modes[0].end(), [this](auto& item) {
+         return (item == hardware_interface::HW_IF_VELOCITY ||
+                 item == hardware_interface::HW_IF_POSITION /*|| item == PASSTHROUGH_GPIO*/);
```

Comment:
Shouldn't this be uncommented?
```suggestion
                 item == hardware_interface::HW_IF_POSITION || item == PASSTHROUGH_GPIO);
```